src/site/content/en/blog/building-a-sidenav-component/index.md

```diff
@@ -0,0 +1,333 @@
+
+<aside id="sidenav-open">
+  …
+</aside>
+```
+
+Clicking each of these links changes the hash state of our page URL, 
+then with a pseudo-class I show and hide the sidenav:
+
+```css
+@media (max-width: 540px) {
+  #sidenav-open {
+    visibility: hidden;
+  }
+
+  #sidenav-open:target {
+    visibility: visible;
+  }
+}
+```
+
+<figure class="w-figure w-figure--fullbleed">
+  <video playsinline controls autoplay loop muted class="w-screenshot">
+    <source src="https://storage.googleapis.com/atoms-sandbox.google.com.a.appspot.com/hash-change.mp4">
+  </video>
+</figure>
+
+### CSS Grid {: #grid-stacks }
+
+In the past, I only used absolute or fixed position 
+sidenav layouts and components. Grid though, with its `grid-area` syntax, 
+lets us assign multiple elements to the same row or column.
+
+#### Stacks
+
+The primary layout element `#sidenav-container` is a grid that creates 1 row and 2 columns, 
+1 of each are named `stack`. When space is constrained, CSS assigns all of the `<main>` element's 
+children to the same grid name, placing all elements into the same space, creating a stack.
+
+```css
+#sidenav-container {
+  display: grid;
+  grid: [stack] 1fr / min-content [stack] 1fr;
+  min-height: 100vh;
+}
+
+@media (max-width: 540px) {
+  #sidenav-container > * {
+    grid-area: stack;
+  }
+}
+```
+
+<figure class="w-figure w-figure--fullbleed">
+  <video playsinline controls autoplay loop muted class="w-screenshot">
+    <source src="https://storage.googleapis.com/atoms-sandbox.google.com.a.appspot.com/responsive-stack-demo-1080p.mp4">
+  </video>
+</figure>
+
+#### Menu backdrop
+
+The `<aside>` is the animating element that contains the side navigation. It has
+2 children: the navigation container `<nav>` named `[nav]` and a backdrop `<a>`
+named `[escape]`, which is used to close the menu. 
+
+```css
+#sidenav-open {
+  display: grid;
+  grid-template-columns: [nav] 2fr [escape] 1fr;
+}
+```
+
+Adjust `2fr` & `1fr` to find the ratio you like for the menu overlay and its negative space close button.
+
+<figure class="w-figure w-figure--fullbleed">
+  <video playsinline controls autoplay loop muted class="w-screenshot">
+    <source src="https://storage.googleapis.com/atoms-sandbox.google.com.a.appspot.com/overlay-escape-ratio.mp4">
+  </video>
+  <figcaption class="w-figure">
+    A demo of what happens when you change the ratio.
+  </figcaption>
+</figure>
+
+### CSS 3D transforms & transitions {: #transforms }
+
+Our layout is now stacked at a mobile viewport size. Until I add some new styles, 
+it's overlaying our article by default. Here's some UX I'm shooting for in this next section:
+
+- Animate open and close
+- Only animate with motion if the user is OK with that
+- Animate `visibility` so keyboard focus doesn't enter the offscreen element
+
+Animate between open and close, `:target` and `:not(:target)`, 
+did someone say animation or motion!? Let's start with accessibility.
```

Comment:
Can you rephrase this? I'm not sure what you mean here.

Reply:
replaced with:

`As I begin to implement motion animations, I want to start with accessibility top of mind.`